tests/unit/sceneImportTests.cpp

```diff
@@ -15,183 +15,178 @@ class TestImporter : public Importer {
 
-    REQUIRE(root["import"].Scalar() == "");
-    REQUIRE(root["name"].Scalar() == "thisCar");
-    REQUIRE(root["type"].Scalar() == "car");
-    REQUIRE(root["category"].Scalar() == "vehicle");
-}
+    // Check that global texture URLs are resolved correctly.
 
-TEST_CASE( "Texture path tests after importing process", "[scene-import][core]") {
-    TestImporter importer;
-    auto root = importer.applySceneImports("scene.yaml");
-
-    REQUIRE(root["import"].Scalar() == "");
-    const auto& texturesNode = root["textures"];
-    REQUIRE(texturesNode["tex1"]["url"].Scalar() == "resources/icons/poi.png");
-    REQUIRE(texturesNode["tex2"]["url"].Scalar() == "/absoluteTexPath/texture.png");
-    REQUIRE(texturesNode["tex3"]["url"].Scalar() == "resources/tex/sameName.png");
-    REQUIRE(texturesNode["tex4"]["url"].Scalar() == "import/../resources/icons/poi2.png");
-    REQUIRE(texturesNode["tex5"]["url"].Scalar() == "import/importResources/tex.png");
-    REQUIRE(texturesNode["tex6"]["url"].Scalar() == "/absPath/tex.png");
-
-    const auto& styleNode = root["styles"]["styleA"];
-    const auto& uniformsNode = styleNode["shaders"]["uniforms"];
-
-    REQUIRE(styleNode["texture"].Scalar() == "resources/tex/sameName.png");
-    REQUIRE(root["styles"]["styleB"]["texture"].Scalar() == "tex5");
-
-    REQUIRE(uniformsNode["u_tex1"].Scalar() == "resources/tex/sameName.png");
-    REQUIRE(uniformsNode["u_tex2"][0].Scalar() == "uTex1.png");
-    REQUIRE(uniformsNode["u_tex2"][1].Scalar() == "resources/uTex2.png");
-    REQUIRE(uniformsNode["u_tex3"].Scalar() == "import/importResources/tex.png");
-    REQUIRE(uniformsNode["u_tex4"].Scalar() == "tex1");
+    auto textures = root["textures"];
+
+    CHECK(textures["tex1"]["url"].Scalar() == "/root/path/to/texture.png");
+    CHECK(textures["tex2"]["url"].Scalar() == "/up_a_directory.png");
+    CHECK(textures["tex3"]["url"].Scalar() == "/root/imports/in_imports.png");
+    CHECK(textures["tex4"]["url"].Scalar() == "/root/not_in_imports.png");
+    CHECK(textures["tex5"]["url"].Scalar() == "/at_root.png");
+
+    // Check that "inline" texture URLs are resolved correctly.
+
+    auto styleA = root["styles"]["styleA"];
+
+    CHECK(styleA["texture"].Scalar() == "/root/path/to/texture.png");
+
+    auto uniformsA = styleA["shaders"]["uniforms"];
+
+    CHECK(uniformsA["u_tex1"].Scalar() == "/at_root.png");
+    CHECK(uniformsA["u_tex2"][0].Scalar() == "/root/path/to/texture.png");
+    CHECK(uniformsA["u_tex2"][1].Scalar() == "tex2");
+    CHECK(uniformsA["u_bool"].Scalar() == "true");
+    CHECK(uniformsA["u_float"].Scalar() == "0.25");
+    CHECK(uniformsA["u_tex3"].Scalar() == "tex3");
+
+    auto styleB = root["styles"]["styleB"];
+
+    CHECK(styleB["texture"].Scalar() == "/root/imports/in_imports.png");
+
+    auto uniformsB = styleB["shaders"]["uniforms"];
+
+    CHECK(uniformsB["u_tex1"].Scalar() == "/root/imports/in_imports.png");
+    CHECK(uniformsB["u_tex2"].Scalar() == "tex2");
```

Comment:
I think this one is not correct: It would mean than the imported scene-node may reference textures from the importing scene-node (or any import level above).